static/js/redux/ui/CompareTimetableSidebar.tsx

```diff
@@ -38,8 +44,42 @@ const CompareTimetableSideBar = () => {
     );
   };
 
-  const activeSlots = activeCourses.map((course) => createMasterSlot(course, 0));
-  const comparedSlots = comparedCourses.map((course) => createMasterSlot(course, 1));
+  const isOfferingInTimetable = (timetable: Timetable, offeringId: number) => {
+    let inTimetable = false;
+    timetable.slots.forEach((currentSlot) => {
+      if (currentSlot.offerings.indexOf(offeringId) !== -1) {
+        inTimetable = true;
+      }
+    });
+    return inTimetable;
+  };
+
+  /**
+   * returns an array of course ids of the courses that have the same section in both timetables
+   */
+  const getSectionsInBothTimetables = () => {
```

Comment:
Do you think you could extract the relevant methods in both CompareTimetableSidebar and SlotManager into a single file (e.g. compareTimetablesUtils)? They are both dependent on the same code, and we don't want to duplicate it (as we might change code in one of the files but not the other).

Reply:
ok I will look into it